src/Pos/Block/Network/Retrieval.hs

```diff
@@ -122,9 +123,10 @@ retrievalWorkerImpl sendActions =
         updateRecoveryHeader nodeId header
         endedRecoveryVar <- newEmptyMVar
         let cont (headers :: NewestFirst NE (BlockHeader ssc)) =
-                let firstHeader = headers ^. _Wrapped . _neLast
+                let oldestHeader = headers ^. _Wrapped . _neLast
```

Comment:
Can you avoid `_Wrapped` please? The issue is that if the type changes to `OldestFirst`, the logic will break, but the code will compile. There's a `_NewestFirst` iso for this.